### PR TITLE
[softmax] Update softmax implementation

### PR DESCRIPTION
**Pull request by kparichay**
Update softmax implementation to use tensor operations than running manual loops

See also #483

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: nntrainer/src/tensor.cpp [hunk 1]
```diff
@@ -239,6 +239,20 @@ Tensor Tensor::add(float const &value) {
   return result;
 }
 
+void Tensor::saxpy(const unsigned int N, const float alpha, const float *X,
+                   const int incX, float *Y, const int incY) {
```
Comment (jijoongmoon) [inline nntrainer/src/tensor.cpp:243]: I think we have to make this dummy func in different file for better managing. Also we have to add more blas func like gemm.

File: nntrainer/src/activation_layer.cpp [hunk 2]
```diff
@@ -148,48 +150,37 @@ Tensor ActivationLayer::softmax(Tensor const &t) {
...
-    }
+  // take exp
+  Tensor result = divisor.apply(exp_util);
+  rp = result.getData();
+  // take sum over batch
+  Tensor sum = result.sum_by_batch();
+
+  for (unsigned int k = 0; k < batch; k++) {
+    int index = k * feat_len;
+    std::transform(rp + index, rp + index + feat_len, rp + index,
```
Comment (jijoongmoon) [inline nntrainer/src/activation_layer.cpp:179]: transform is very helpful.

**Review by jijoongmoon — APPROVED** [review]
LGTM

File: nntrainer/src/tensor.cpp (hunk 1 above)
Comment (kparichay) [reply nntrainer/src/tensor.cpp:243]: yes.

**Pull request merged**